src/hooks/useImportingWallet.js

```diff
@@ -261,18 +267,28 @@ export default function useImportingWallet({ showImportModal = true } = {}) {
             image
           )
             .then(success => {
-              handleSetImporting(false);
+              ios && handleSetImporting(false);
               if (success) {
                 goBack();
                 InteractionManager.runAfterInteractions(async () => {
                   if (previousWalletCount === 0) {
-                    replace(Routes.SWIPE_LAYOUT, {
+                    const action = ios ? replace : navigate;
+                    action(Routes.SWIPE_LAYOUT, {
                       params: { initialized: true },
                       screen: Routes.WALLET_SCREEN,
                     });
                   } else {
                     navigate(Routes.WALLET_SCREEN, { initialized: true });
                   }
+                  if (android) {
+                    handleSetImporting(false);
+                    InteractionManager.runAfterInteractions(() =>
+                      setIsWalletLoading(null)
+                    );
+                    dispatchNavigation({
+                      type: '@RAINBOW/REMOVE_FIRST',
+                    });
```

Comment:
Can we move this entire thing to Navigation? I guess this is the place it should be